lib/bunyan.js

```diff
@@ -8,209 +18,124 @@
  * vim: expandtab:ts=4:sw=4
  */
 
-var VERSION = '1.8.10';
+const VERSION = '1.8.10';
 
 /*
  * Bunyan log format version. This becomes the 'v' field on all log records.
  * This will be incremented if there is any backward incompatible change to
  * the log record format. Details will be in 'CHANGES.md' (the change log).
  */
-var LOG_VERSION = 0;
+const LOG_VERSION = 0;
 
+const INTERNAL_DEBUG = false; // Set this to true to enable internal dev/logging.
 
-var xxx = function xxx(s) {     // internal dev/debug logging
-    var args = ['XX' + 'X: '+s].concat(
-        Array.prototype.slice.call(arguments, 1));
-    console.error.apply(this, args);
-};
-var xxx = function xxx() {};  // comment out to turn on debug logging
-
+let xxx, safeJsonStringify, mv, RotatingFileStream;
 
-/*
- * Runtime environment notes:
- *
- * Bunyan is intended to run in a number of runtime environments. Here are
- * some notes on differences for those envs and how the code copes.
- *
- * - node.js: The primary target environment.
- * - NW.js: http://nwjs.io/  An *app* environment that feels like both a
- *   node env -- it has node-like globals (`process`, `global`) and
- *   browser-like globals (`window`, `navigator`). My *understanding* is that
- *   bunyan can operate as if this is vanilla node.js.
- * - browser: Failing the above, we sniff using the `window` global
- *   <https://developer.mozilla.org/en-US/docs/Web/API/Window/window>.
- *      - browserify: http://browserify.org/  A browser-targetting bundler of
- *        node.js deps. The runtime is a browser env, so can't use fs access,
- *        etc. Browserify's build looks for `require(<single-string>)` imports
- *        to bundle. For some imports it won't be able to handle, we "hide"
- *        from browserify with `require('frobshizzle' + '')`.
- * - Other? Please open issues if things are broken.
- */
-var runtimeEnv;
-if (typeof (process) !== 'undefined' && process.versions) {
-    if (process.versions.nw) {
-        runtimeEnv = 'nw';
-    } else if (process.versions.node) {
-        runtimeEnv = 'node';
-    }
-}
-if (!runtimeEnv && typeof (window) !== 'undefined' &&
-    window.window === window) {
-    runtimeEnv = 'browser';
-}
-if (!runtimeEnv) {
-    throw new Error('unknown runtime environment');
+if (INTERNAL_DEBUG) {
+  xxx = (s) => { // internal dev/debug logging
+    const args = ['XXX: ' + s].concat(Array.prototype.slice.call(arguments, 1));
```

Comment:
The arguments object is not created in the arrow function.
This argument refers to arguments outside the arrow function.
If you want to retrieve the arguments of an arrow function, you need to use Rest Parameter instead of using the arguments object.

Reply:
This function mostly likely will be removed later on, so it's optional to fix; though it can be a good learning opportunity.

Reply:
@curbengh @segayuu I prefer just remove internal debug comepeletely.

Reply:
suggest to do it in a separate PR.